src/components/CountrySearchSelectInput/index.tsx

```diff
@@ -0,0 +1,179 @@
+                                        if (isFalsyString(potentialCountry.society_name)) {
+                                            return undefined;
+                                        }
+
+                                        return {
+                                            ...potentialCountry,
+                                            id: potentialCountry.id,
+                                            name: potentialCountry.society_name,
+                                        };
+                                    }
+
+                                    if (isFalsyString(potentialCountry.name)) {
+                                        return undefined;
+                                    }
+
+                                    return {
+                                        ...potentialCountry,
+                                        id: potentialCountry.id,
+                                        name: potentialCountry.name,
+                                    };
+                                },
+                            ).filter(isDefined) ?? [],
+                            ...prevOptions,
+                        ],
+                        (country) => country.id,
+                    )
+                ),
+            );
+        },
+        [selectionMode],
+    );
+
+    const { trigger: getDetailsForSelectedCountry } = useLazyRequest<'/api/v2/country/{id}/', null>({
+        url: '/api/v2/country/{id}/',
+        pathVariables: isDefined(value) ? { id: value } : undefined,
+        onSuccess: (response) => {
+            addCachedOptions([response]);
+        },
+    });
+
+    const {
+        pending: searchPending,
+        response: searchResponse,
+    } = useRequest({
+        skip: searchTextSafe.length < MIN_SEARCH_LENGTH,
+        url: '/api/v2/country/',
+        query: {
+            search: searchTextSafe,
+            limit: 20,
+            is_independent: true,
+            is_deprecated: false,
+            is_nationalsociety: selectionMode === 'nationalSociety' ? true : undefined,
+        },
+        onSuccess: (response) => {
+            addCachedOptions(response?.results);
+        },
+    });
+
+    useEffect(
+        () => {
+            if (isNotDefined(value)) {
+                return;
+            }
+
+            const countryDetail = cachedOptions.find((option) => option.id === value);
+            if (!countryDetail) {
+                getDetailsForSelectedCountry(null);
+            }
+        },
+        [value, cachedOptions, getDetailsForSelectedCountry],
+    );
+
+    return (
+        <SearchSelectInput
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...otherProps}
+            className={className}
+            name={name}
+            onSearchValueChange={setSearchText}
+            optionsPending={searchPending}
+            searchOptions={cachedOptions}
+            options={cachedOptions}
```

Comment:
It's easier if we use different values here.

Update options only when the option is selected.
Get searchOptions always from the request.